pkg/providers/vsphere/vmlifecycle/bootstrap_sysprep.go

```diff
@@ -160,6 +160,25 @@ func convertTo(from *vmopv1sysprep.Sysprep, bsArgs *BootstrapArgs) *vimtypes.Cus
 		sysprepCustomization.LicenseFilePrintData = convertLicenseFilePrintDataTo(from.LicenseFilePrintData)
 	}
 
+	if pkgcfg.FromContext(vmCtx).Features.GuestCustomizationVCDParity {
+		if bootstrapData.Sysprep != nil {
+			sysprepCustomization.ScriptText = bootstrapData.Sysprep.ScriptText
+		}
+
+		if from.ExpirePasswordAfterNextLogin {
+			sysprepCustomization.ResetPassword = vimtypes.NewBool(true)
+		}
+
+		if id, ok := vmCtx.VM.Annotations[pkgconst.VCFAIDAnnotationKey]; ok {
```

Comment:
Please rewrite as `if id := vmCtx.VM.Annotations[pkgconst.VCFAIDAnnotationKey]; id != "" {`. Unlike real ExtraConfig, there is no real reason to allow an empty value to clear anything.